checks/cassandra/check.go

```diff
@@ -0,0 +1,45 @@
+package cassandra
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gocql/gocql"
+)
+
+// Config is the Cassandra checker configuration settings container.
+type Config struct {
+	// Hosts is a list of Cassandra hosts. At least one is required.
+	Hosts []string
+	// Keyspace is the Cassandra keyspace to which you want to connect. Required.
+	Keyspace string
+}
+
+// New creates new Cassandra health check that verifies the following:
+// - that a connection can be established through creating a session
+// - that queries can be executed by describing keyspaces
+func New(config Config) func(ctx context.Context) error {
+	return func(ctx context.Context) (checkErr error) {
+		if len(config.Hosts) < 1 || len(config.Keyspace) < 1 {
+			checkErr = errors.New("keyspace name and hosts are required to initialize cassandra health check")
+		}
+
+		cluster := gocql.NewCluster(config.Hosts...)
+		cluster.Keyspace = config.Keyspace
+
+		session, err := cluster.CreateSession()
+		if err != nil {
+			checkErr = fmt.Errorf("cassandra health check failed on connect: %w", err)
+			return
+		}
+		defer session.Close()
+
+		err = session.Query("DESCRIBE KEYSPACES;").WithContext(ctx).Exec()
+		if err != nil {
+			checkErr = fmt.Errorf("cassandra health check failed on describe: %w", err)
+			return
+		}
+
+		return
```

Comment:
since this check does not have any possible deferred error, please remove the named return and return the errors immediately instead of assigning to the returned variable.

Reply:
Can do!